open_bus_stride_api/routers/gtfs_rides_agg.py

```diff
@@ -36,49 +37,63 @@ class GtfsRidesAggGroupByPydanticModel(pydantic.BaseModel):
 PYDANTIC_MODEL = GtfsRidesAggPydanticModel
 GROUP_BY_PYDANTIC_MODEL = GtfsRidesAggGroupByPydanticModel
 DEFAULT_LIMIT = 1000
-ALLOWED_GROUP_BY_FIELDS = ['gtfs_route_date', 'operator_ref', 'day_of_week']
+ALLOWED_GROUP_BY_FIELDS = ['gtfs_route_date', 'gtfs_route_hour', 'operator_ref', 'day_of_week']
 
 
 @common.router_list(router, TAG, PYDANTIC_MODEL, WHAT_PLURAL)
 def list_(limit: int = common.param_limit(default_limit=DEFAULT_LIMIT),
           offset: int = common.param_offset(),
           get_count: bool = common.param_get_count(),
           date_from: datetime.date = common.doc_param('date', filter_type='date_from', default=...),
-          date_to: datetime.date = common.doc_param('date', filter_type='date_to', default=...)):
-    sql = dedent("""
+          date_to: datetime.date = common.doc_param('date', filter_type='date_to', default=...),
+          exclude_hours_from: int = common.doc_param('hour', filter_type='hour_from', description="Hours to exclude from search, currently used to filter out edge cases."),
+          exclude_hours_to: int = common.doc_param('hour', filter_type='hour_to', description="Hours to exclude from search, currently used to filter out edge cases.")):
+    sql = """
         select
             agg.gtfs_route_id,
-            agg.gtfs_route_date,
+            agg.gtfs_route_hour,
             agg.num_planned_rides,
             agg.num_actual_rides,
             rt.operator_ref
-        from gtfs_rides_agg agg, gtfs_route rt
+        from gtfs_rides_agg_by_hour agg, gtfs_route rt
         where
             agg.gtfs_route_id = rt.id
-            and agg.gtfs_route_date >= :date_from
-            and agg.gtfs_route_date <= :date_to
-    """)
+            and date_trunc('day', agg.gtfs_route_hour) >= :date_from
+            and date_trunc('day', agg.gtfs_route_hour) <= :date_to   
```

Comment:
in case the user wants to get everything from 10PM to 3AM, will they need two requests?